pkg/controllers/terminating/v1alpha1/terminate.go

```diff
@@ -38,105 +38,58 @@ type Terminator struct {
 	coreV1Client  corev1.CoreV1Interface
 }
 
-// cordonNodes takes in a list of expired nodes as input and cordons them
-func (t *Terminator) cordonNodes(ctx context.Context) error {
-	// 1. Get terminable nodes
-	nodeList, err := t.getLabeledNodes(ctx, provisioning.ProvisionerTerminablePhase)
-	if err != nil {
-		return err
-	}
-	// 2. Cordon nodes
-	for _, node := range nodeList {
-		persisted := node.DeepCopy()
-		node.Spec.Unschedulable = true
-		node.Labels = functional.UnionStringMaps(
-			node.Labels,
-			map[string]string{provisioning.ProvisionerPhaseLabel: provisioning.ProvisionerDrainingPhase},
-		)
-		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
-			return fmt.Errorf("patching node %s, %w", node.Name, err)
-		}
-		zap.S().Debugf("Cordoned node %s", node.Name)
+// cordonNode cordons a node
+func (t *Terminator) cordonNode(ctx context.Context, node *v1.Node) error {
+	persisted := node.DeepCopy()
+	node.Spec.Unschedulable = true
+	if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
```

Comment:
Is it worth checking on whether it's already Unschedulable in which case no need to talk to apiserver?

Reply:
(or does `Patch()` already do that?)

Reply:
Ellis and I discussed this in the other PR for termination watch events (since this code was pretty similar).
https://github.com/awslabs/karpenter/pull/446#discussion_r650227408

In the controller-runtime patch code, it says that the MergeFrom computes the diff, so the Patch will check that for us.
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/patch.go#L33